api/envoy/config/filter/network/thrift_proxy/v2alpha1/route.proto

```diff
@@ -66,8 +70,42 @@ message RouteMatch {
   repeated envoy.api.v2.route.HeaderMatcher headers = 4;
 }
 
-// [#comment:next free field: 2]
+// [#comment:next free field: 3]
 message RouteAction {
-  // Indicates the upstream cluster to which the request should be routed.
-  string cluster = 1 [(validate.rules).string.min_bytes = 1];
+  oneof cluster_specifier {
+    option (validate.required) = true;
+
+    // Indicates a single upstream cluster to which the request should be routed
+    // to.
+    string cluster = 1 [(validate.rules).string.min_bytes = 1];
+
+    // Multiple upstream clusters can be specified for a given route. The
+    // request is routed to one of the upstream clusters based on weights
+    // assigned to each cluster.
+    WeightedCluster weighted_clusters = 2;
+  }
+}
```

Comment:
is there any chance that this can be moved to TCP proxy in general and you use the thrift as read only decoder (just like mongo)? It has the benefit of being universally applicable across all tcp stuff with the exception of redis.

Reply:
or is this a separate filter stack for thrift?

Reply:
yeah it's a separate filter stack. see [here](https://github.com/envoyproxy/envoy/blob/a8572192f2e59eaa63361545d4c32371de1fc320/test/extensions/filters/network/thrift_proxy/integration_test.cc#L23) for a sample configuration